adoptopenjdk-api-v3-frontend/src/main/kotlin/net/adoptopenjdk/api/v3/Pagination.kt

```diff
@@ -0,0 +1,23 @@
+package net.adoptopenjdk.api.v3
+
+import kotlin.math.min
+
+object Pagination {
+    private const val defaultPageSizeNum = 10
+    private const val maxPageSizeNum = 500
```

Comment:
Max size of 500 may be excessive, I estimate that would set up a 4.5Mb download, I think 50 may be more reasonable

Reply:
In fact given the current max is 20, can go with 20 to maintain consistency

Reply:
@johnoliver Is this really a problem? The default would still be 10 items per page and unless people override the page size, this wouldn't change the current behavior.

I would go with a maximum of 100 items per page, if that's okay. Otherwise we wouldn't have to merge this PR at all.
Its purpose is, after all, to enable page sizes larger than 20 items. 😉

Reply:
I viewed the main benefit of this PR about adding the page limit to the Release List resource and documenting the maximum, not to increase the maximum. Although it was incorrectly not documented that there was a limit of 20, it was intended for 20 to be the maximum. I am not sure if there is a usecase I want to support that would allow people to access 100 releases with a single request. In the case of the release_names given it is just returning strings I agree that a higher limit is fine, so if you want to have 100 or even 500 on that endpoint, fine, but for the other endpoint 100 releases is still a lot for a single request.

Reply:
@johnoliver Just so I understand the restrictions properly:
Is it about the size of the HTTP response and the resources the application requires to serialize the payload, to reduce the load on the database, or to prevent "sloppy" (misbehaving) clients from fetching too many items at once?

I can speak for my use case, which is always fetching the complete release list and avoid making more HTTP request roundtrips than necessary in https://github.com/joschi/java-metadata/blob/master/bin/adoptopenjdk.bash#L119-L127.